core/src/Streamly/Internal/Data/Fold/Container.hs

```diff
@@ -326,6 +326,10 @@ demuxGeneric getKey getFold = fmap extract $ foldlM' step initial
 -- This can be used to scan a stream and collect the results from the scan
 -- output.
 --
+-- /Warning/: One should call the returned monadic action to make sure the
+-- folds fully complete—even if the action returns nothing useful (e.g., a map
+-- of @()@ values).
```

Comment:
This should not be the case unless there is some laziness somewhere. Can you test and confirm that this makes a difference?